packages/framework/aqueduct/src/container-runtime-factories/baseContainerRuntimeFactory.ts

```diff
@@ -91,6 +92,48 @@ export class BaseContainerRuntimeFactory implements
         return runtime;
     }
 
+    public async instantiateFirstTime(context: IContainerContext): Promise<IRuntime> {
```

Comment:
This is interesting example of base class - it has containerHasInitialized, as well as containerInitializingFirstTime. Note that it does not have the opposite (containerInitializingFromExisting), assuming it's not needed (which is fishy assumption IMHO).